### ISSUE TITLE
Version 3.3.4 breaks support for Python versions earlier than 3.9

### ISSUE THREAD

**Issue by zivnevo** [qntm/greenery#59]
The reason is in `fsm.py`, line 59:
```
finals: set[state_type]
```
In Python versions earlier than 3.9 this causes an error:
```
TypeError: 'type' object is not subscriptable
```
This is because [Type Hinting Generics in Standard Collections was only introduced in Python 3.9](https://docs.python.org/3/whatsnew/3.9.html#type-hinting-generics-in-standard-collections).

**Comment by qntm** [comment]
@Seanny123 Any comment on this?

**Comment by Seanny123** [comment]
Yep, this is my fault. I should have instead used `from typing import Set` and `finals: Set[state_type]` to maintain compatibility with older versions of Python.

**Comment by qntm** [comment]
Should be resolved in `greenery` 3.3.5. Thank you @Seanny123 for explaining how to fix this.
